packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -395,14 +395,17 @@ export class CdkToolkit {
   private async selectStacksForDiff(stackNames: string[], exclusively?: boolean) {
     const assembly = await this.assembly();
 
-    const idsToValidate = process.env.STACKS_TO_VALIDATE ? process.env.STACKS_TO_VALIDATE.split(';') : stackNames;
-    const stacksToValidate = await this.selectStacksForList(idsToValidate);
-    await this.validateStacks(stacksToValidate);
+    const allStacks = await this.selectStacksForList([]);
+    await this.validateStacks(allStacks.filter(art => art.validateOnSynth ?? false));
```

Comment:
Preferably we'd validate all stacks in one `validateStacks()` call. The difference is a more complete view of all errors, as synthesis will stop after this call if it encounters errors (and so wouldn't show any errors that would be encountered in the `validateStacks()` call below)